odl/test/solvers/functional/default_functionals_test.py

```diff
@@ -614,5 +614,73 @@ def test_weighted_proximal_L1_norm_close(space):
     assert all_almost_equal(expected_result, p_ip)
 
 
+def test_bregman_functional_no_gradient(space):
+    """Test Bregman distance for functional without gradient.
+
+    Test that the Bregman distance functional fails if the underlying
+    functional does not have a gradient and no subgradient operator is
+    given. Also test giving the subgradient operator separately.
+    """
+
+    ind_func = odl.solvers.IndicatorNonnegativity(space)
+    point = noise_element(space)
+
+    # Indicator function has no gradient, hence one cannot create a bregman
+    # distance functional
+    with pytest.raises(NotImplementedError):
+        odl.solvers.BregmanDistance(ind_func, point)
```

Comment:
We need to do tests for both `functiona.bregman` and the pure functional `odl.solvers.BregmanDistance`, at least one that verifies that it points to the later I'd say.

Reply:
My idea, when keeping these test after implementing the `functional.bregman`, was that they still test the `BregmanDistance` class. Not for all functionals, but  some kind of test for the class itself.

I also added a test for `functional.bregman` to the suite of tests for functionals, [here](https://github.com/odlgroup/odl/pull/1267/files/3f2a8c2dfb158a059a214858b3d5ecce8bb640a9#diff-18b72305369f0cf5b0f7408c5211a60dR581). My idea was that this should cover all available functionals and see that there `.bregman` works as expected. By that we also cover the case if one does a special optimized implementation of the Bregman distance for some functionals later (e.g., the squared L2 norm should just be a translated functional).

Is that good enough, or would you like to add some other tests or change some of the current ones?